app/utils/server/functions/normalizeMessageFileUpload.js

```diff
@@ -2,9 +2,13 @@ import { getURL } from '../../lib/getURL';
 import { FileUpload } from '../../../file-upload/server';
 import { Uploads } from '../../../models/server';
 
+function userId(message) {
+	return message.u ? message.u._id : message.agentId;
```

Comment:
Since the `normalizeMessageFileUpload` is a public method, I mean, it's not a specific Livechat method, I don't like the idea of testing a field which is only used by a specific feature, such as the Livechat. Did you get it?

My suggestion is, instead of providing the `agentId` field here, what about providing the `message.u.` property before calling the `normalizeMessageFileUpload` [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/server/hooks/sendToCRM.js#L64)?
It would look like this:
```
    const u = message;
    postData.messages.push(normalizeMessageFileUpload({ u, ...msg }));
```

What do you think? 
Thanks.

Reply:
You are right, seems a lot better that way. I'll do it